graph/patterns/namespace.md

```diff
@@ -0,0 +1,125 @@
+# Namespace
+
+Microsoft Graph API Design Pattern
+
+### *The Namespace provides the ability to organize resource definitions together into a logical set.*
+
+## Problem
+
+When building a complex offering API designers may need to model many different
+resources and their relationships. For better user experience and
+discoverability related API elements need to be grouped together.  
+
+
+## Solution
+
+API designers can use the Namespace attribute of the CSDL schema to declare a
+namespace and logically organize related API entities in the Graph metadata.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ XML
+<Schema Namespace="microsoft.graph.{namespace}">
+...
+</Schema>
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A public namespace must have "microsoft.graph.” prefix and be presented in camel
+case, i.e microsoft.graph.myNamespace.
+
+When type casting is required in the API query, request or response, a fully
+qualified type name is represented as concatenation of a namespace and a type
+name. For consistent user experience namespaces should be aligned with the corresponding API category path segment.
+
+
+## When to Use this Pattern
+
+API resource grouping creates a user-friendly experience keeping all resources
+for a specific feature close together. It also allows generating smallersmaller downloadable client components 
```

Comment:
@darrelmiller should we be mentioning smaller downloadable client components?  Is that really true?